app/pages/time/index.page.tsx

```diff
@@ -23,6 +33,33 @@ const Time = (): JSX.Element => {
 
   const [selectedDate, setSelectedDate] = useState(new Date())
 
+ const router = useRouter()
+
+  const urldate = router.query.date?.toString()
+
+  const CurrentDate= 
+ urldate ? 
+parse(urldate, 'yyyy-MM-dd', new Date())
+:
+new Date()
+
+const StartOfWeek = startOfWeek(CurrentDate,{weekStartsOn:1} )
+const EndOfWeek = endOfWeek(CurrentDate, {weekStartsOn: 1})
+
+
+const startDate = format(StartOfWeek, "	dd.MM")
+const endDate = format(EndOfWeek, "dd.MM.yyyy")
+
+
+
+ const nextWeek = addDays(CurrentDate,7)
+ const previousWeek = addDays(CurrentDate, -7)
+ const previousWeekString = format(new Date(previousWeek), 'yyyy-MM-dd')
+ const nextWeekString = format(new Date(nextWeek), 'yyyy-MM-dd')
```

Comment:
`previousWeek` and `nextWeek` are already `Date` objects. You can use them directly without `new Date(...)`